assets/components/PersonalizeHomeModal.tsx

```diff
@@ -0,0 +1,356 @@
+        };
+
+        this.props.saveUser({dashboards: [newDashboard]});
+
+        this.setState({isLoading: false});
+    }
+
+    getSelectedTopics() {
+        const topics: Array<ITopic> = [];
+
+        this.state.selectedTopicIds.map((topicId) => this.wireTopicsById.get(topicId)).forEach((topic) => {
+            if (topic != null) {
+                topics.push(topic);
+            }
+        });
+
+        return topics;
+    }
+
+    render() {
+        const NoSearchMatches = () => (
+            <div className="empty-state__container mt-3">
+                <div className="empty-state">
+                    <figure className="empty-state__graphic">
+                        <img src="/static/empty-states/empty_state--small.svg" role="presentation" alt="" />
+                    </figure>
+                    <figcaption className="empty-state__text">
+                        <h4 className="empty-state__text-heading">{gettext('You don\'t have any saved Topics yet.')}</h4>
+                        <p className="empty-state__text-description">
+                            {gettext('You can create Topics by saving search terms and/or filters from the Wire and Agenda sections.')}
+                        </p>
+                        <div className="empty-state__links">
+                            <a className="" href="/wire">{gettext('Go to Wire')}</a>
+                            <a className="" href="/agenda">{gettext('Go to Agenda')}</a>
+                        </div>
+                    </figcaption>
+                </div>
+            </div>
+        );
+
+        const NoSelectedTopics = () => (
+            <div className="empty-state__container empty-state__container--full-height">
+                <div
+                    style={{
+                        display: 'flex',
+                        flexDirection: 'column',
+                        alignItems: 'center',
+                    }}
+                    className="empty-state empty-state--large"
+                >
+                    <figure className="empty-state__graphic">
+                        <img src="/static/empty-states/empty_state--small.svg" role="presentation" alt="" />
+                    </figure>
+                    <figcaption className="empty-state__text">
+                        <h4 className="empty-state__text-heading">{gettext('You don\'t have any saved Topics yet')}</h4>
+                        <p className="empty-state__text-description">
+                            {gettext('You can create Topics by saving search terms and/or filters from the Wire and Agenda sections.')}
+                        </p>
+                    </figcaption>
+                </div>
+            </div>
+        );
+
+        const filteredTopics = (topics: Array<ITopic>) =>
+            topics.filter(({is_global}) => this.state.activeSection === '1' ? is_global != true : is_global === true);
+
+        const searchMatches = filteredTopics(this.wireTopics ?? [])
```

Comment:
I see you apply a fallback to an empty array here, but not on line 190. Either we should type it as

```
wireTopics: Array<ITopic> | null;
```

or apply the default in the constructor and keep the type non-nullable.